pytorch_lightning/plugins/environments/lightning_environment.py

```diff
@@ -40,13 +40,13 @@ def __init__(self):
         self._global_rank: int = 0
         self._world_size: int = 1
 
-    def creates_children(self) -> bool:
-        """Returns whether the cluster creates the processes or not.
+        # If at least :code:`LOCAL_RANK` is available as environment variable, Lightning assumes the user acts as the
+        # process launcher/job scheduler and Lightning will not launch new processes.
+        self._creates_children: bool = "LOCAL_RANK" in os.environ
 
-        If at least :code:`LOCAL_RANK` is available as environment variable, Lightning assumes the user acts as the
-        process launcher/job scheduler and Lightning will not launch new processes.
-        """
-        return "LOCAL_RANK" in os.environ
+    def creates_children(self) -> bool:
+        """Returns whether the cluster creates the processes or not."""
+        return self._creates_children
```

Comment:
@awaelchli - I'm not fully confident in this change. Could you take another look at this PR whenever you get a chance?

Reply:
The logic was the following. There are two  ways to start ddp with Lightning: 

**Case 1 (manual or like a scheduler does it):**
```
LOCAL_RANK=0 python train.py
LOCAL_RANK=1 python train.py
```

In both processes, `creates_children=True`, so we are interpreting that a "cluster" is starting the processes individually. 

**Case 2 (Lightning  launches the processes):**
python train.py

Here, `creates_children=False`, so this current process becomes LOCAL_RANK 0, and it launches N-1 other processes:

LOCAL_RANK=1 python train.py

These N-1 processes however now get LOCAL_RANK=X in the environment so their value is `creates_children=True` so no more processes get spawned. 

So regarding these use cases, your change is fine because the value of `creates_children` can be determined at the time of instantiation and then never change. So LGTM.